clients/src/main/java/org/apache/kafka/clients/admin/AdministrativeClient.java

```diff
@@ -0,0 +1,628 @@
+     * @throws InterruptedException     If the thread receives an InterruptedException before finishing.
+     * @throws TimeoutException         If the thread times out before receiving a result.
+     * @throws KafkaException           If there was an error deleting the topic.
+     */
+    public void deleteTopic(String topicName, EnumSet<DeleteTopicsFlags> flags) throws InterruptedException {
+        HashSet<String> set = new HashSet<>();
+        set.add(topicName);
+        Map<String, KafkaException> results = deleteTopics(set, flags);
+        KafkaException exception = results.get(topicName);
+        if (exception != null)
+            throw exception;
+    }
+
+    public enum ListTopicsFlags {
+        /**
+         * List internal topics.
+         */
+        INCLUDE_INTERNAL,
+    }
+
+    public static class TopicInfo {
+        private final String name;
+        private final boolean internal;
+        private final Map<Integer, Try<TopicPartitionInfo>> partitions;
+
+        TopicInfo(String name, boolean internal, Map<Integer, Try<TopicPartitionInfo>> partitions) {
+            this.name = name;
+            this.internal = internal;
+            this.partitions = partitions;
+        }
+
+        public String name() {
+            return name;
+        }
+
+        public boolean internal() {
+            return internal;
+        }
+
+        public Map<Integer, Try<TopicPartitionInfo>> partitions() {
+            return partitions;
+        }
+    }
+
+    public static class TopicPartitionInfo {
+        private final int partition;
+        private final Node leader;
+        private final List<Node> replicas;
+        private final List<Node> isr;
+
+        TopicPartitionInfo(int partition, Node leader, List<Node> replicas, List<Node> isr) {
+            this.partition = partition;
+            this.leader = leader;
+            this.replicas = replicas;
+            this.isr = isr;
+        }
+
+        public int partition() {
+            return partition;
+        }
+
+        public Node leader() {
+            return leader;
+        }
+
+        public List<Node> replicas() {
+            return replicas;
+        }
+
+        public List<Node> isr() {
+            return isr;
+        }
+    }
+
+    /**
+     * List topics in the cluster.
+     *
+     * @param flags                     The flags to use on the request.
+     * @return                          A map from topic names to TopicInfo objects or errors.
+     *                                  An error is returned when we are not authorized to describe the given topic.
+     * @throws InterruptedException     If the thread receives an InterruptedException before finishing.
+     * @throws TimeoutException         If the thread times out before receiving cluster metadata.
+     */
+    public Map<String, Try<TopicInfo>> listTopics(EnumSet<ListTopicsFlags> flags) throws InterruptedException {
```

Comment:
should include a `listTopics` that takes a `Collection<String>` of topics to query